src/Sulu/Bundle/ContentBundle/Preview/PreviewMessageHandler.php

```
@@ -0,0 +1,307 @@
+        Registry $registry,
+        ContentMapperInterface $contentMapper,
+        LoggerInterface $logger
+    ) {
+        $this->preview = $preview;
+        $this->logger = $logger;
+        $this->requestAnalyzer = $requestAnalyzer;
+        $this->registry = $registry;
+        $this->contentMapper = $contentMapper;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(ConnectionInterface $conn, array $message, MessageHandlerContext $context)
+    {
+        // reconnect mysql
+        $this->reconnect();
+
+        try {
+            return $this->execute($conn, $context, $message);
+        } catch (\Exception $e) {
+            // send fail message
+            $conn->send(
+                json_encode(
+                    array(
+                        'command' => 'fail',
+                        'code' => $e->getCode(),
+                        'msg' => $e->getMessage(),
+                        'parentMsg' => $message
+                    )
+                )
+            );
+        }
+    }
+
+    /**
+     * Executes command
+     * @param ConnectionInterface $conn
+     * @param MessageHandlerContext $context
+     * @param array $msg
+     * @return mixed|null
+     * @throws PreviewNotStartedException
+     * @throws MissingParameterException
+     */
+    private function execute(ConnectionInterface $conn, MessageHandlerContext $context, $msg)
+    {
+        if (!array_key_exists('command', $msg)) {
+            throw new MissingParameterException('command');
+        }
+        $command = $msg['command'];
+        $result = null;
+
+        switch ($command) {
+            case 'start':
+                $result = $this->start($conn, $context, $msg);
+                break;
+            case 'stop':
+                $result = $this->stop($conn, $context);
+                break;
+            case 'update':
+                $result = $this->update($conn, $context, $msg);
+                break;
```

Should we throw an Exception if the command is not known?

````
switch ..

    default:
        throw new \InvalidArgumentException(sprintf('Command "%s" not known', $command));
````